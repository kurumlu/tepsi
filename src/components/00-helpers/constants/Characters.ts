export const COUNTER_OVERFLOW = '+';
export const URL_PARAMETER_SEARCH = 'q';
export const QUERY_STRING_PREFIX = '?';
export const QUERY_STRING_SEPARATOR = '&';
