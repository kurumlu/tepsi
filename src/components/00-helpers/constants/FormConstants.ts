export const PASSWORD_MIN_LENGTH = 6;

export const PASSWORD_INDICATOR = {
  checks: [
    {
      re: /[a-z]/,
      score: 1,
    },
    {
      re: /[A-Z]/,
      score: 5,
    },
    {
      re: /([a-z].*[A-Z])|([A-Z].*[a-z])/,
      score: 2,
    },
    {
      re: /(.*[0-9].*[0-9].*[0-9])/,
      score: 7,
    },
    {
      re: /.[!@#$%^&*?_~]/,
      score: 5,
    },
    {
      re: /(.*[!@#$%^&*?_~].*[!@#$%^&*?_~])/,
      score: 7,
    },
    {
      re: /([a-zA-Z0-9].*[!@#$%^&*?_~])|([!@#$%^&*?_~].*[a-zA-Z0-9])/,
      score: 3,
    },
    {
      re: /(.)\1+$/,
      score: 2,
    },
  ],
  options: {
    passIndex: 2,
    label: 'password.indicator.strength.label',
    verdicts: [
      'password.indicator.strength.short',
      'password.indicator.strength.very.weak',
      'password.indicator.strength.weak',
      'password.indicator.strength.good',
      'password.indicator.strength.strong',
      'password.indicator.strength.very.strong',
    ],
    colors: ['#d90c0c', '#d90c0c', '#d90c0c', '#feae3a', '#1ebc13', '#1ebc13'],
    width: ['0%', '15%', '30%', '60%', '80%', '100%'],
    scores: [10, 15, 25, 45],
    bannedPass: 'Not allowed',
    render: true,
  },
  bannedPasswords: [
    '123456',
    '12345',
    '123456789',
    'password',
    'iloveyou',
    'princess',
    'rockyou',
    '1234567',
    '12345678',
    'abc123',
    'nicole',
    'daniel',
    'babygirl',
    'monkey',
    'jessica',
    'lovely',
    'michael',
    'ashley',
    '654321',
    'qwerty',
    'password1',
    'welcome',
    'welcome1',
    'password2',
    'password01',
    'password3',
    'p@ssw0rd',
    'passw0rd',
    'password4',
    'password123',
    'summer09',
    'password6',
    'password7',
    'password9',
    'password8',
    'welcome2',
    'welcome01',
    'winter12',
    'spring2012',
    'summer12',
    'summer2012',
  ],
};
