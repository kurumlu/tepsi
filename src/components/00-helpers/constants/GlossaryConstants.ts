export const GLOSSARY_HIGHLIGHT_CLASS = 'glossary-highlight';
