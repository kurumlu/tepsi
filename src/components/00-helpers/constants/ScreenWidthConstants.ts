export const MIN_SCREEN_WIDTH_DESKTOP_LARGE = 1400;
export const MIN_SCREEN_WIDTH_DESKTOP = 1025;
export const MIN_SCREEN_WIDTH_TABLET = 768;
export const MAX_SCREEN_WIDTH_TABLET = 1024;
export const MIN_SCREEN_WIDTH_TABLET_SMALL = 526;
export const MAX_SCREEN_WIDTH_TABLET_SMALL = 767;
export const MIN_SCREEN_WIDTH_MOBILE_LARGE = 401;
export const MAX_SCREEN_WIDTH_MOBILE_LARGE = 525;
export const MAX_SCREEN_WIDTH_MOBILE = 400;
