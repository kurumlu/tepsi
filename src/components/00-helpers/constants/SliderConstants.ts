import { StringNumberMap } from '../../../common-types';

export const SLIDE_SPACING: StringNumberMap = {
  MOBILE: 16,
  TABLET: 24,
  DESKTOP: 32,
};

export const SLIDES_AMOUNT_DESKTOP = 4;
export const SLIDES_AMOUNT_DESKTOP_LARGE = 5;
