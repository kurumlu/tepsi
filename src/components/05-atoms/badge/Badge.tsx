import * as React from 'react';
import { FunctionComponent } from 'react';
import cn from 'classnames';
import { COUNTER_OVERFLOW } from '../../00-helpers/constants/Characters';

export enum BadgeColor {
  SECONDARY = 'secondary',
  SUBTLE = 'subtle',
}

export enum BadgeSize {
  LARGE = 'l',
}

export type BadgeProps = {
  count: number;
  max?: number;
  color?: BadgeColor;
  size?: BadgeSize;
  className?: string;
  dataQA?: string;
};

export const Badge: FunctionComponent<BadgeProps> = ({
  count,
  max = 99,
  color,
  size,
  className,
  dataQA,
}) => (
  <div
    className={cn(
      'as-a-badge',
      {
        [`as-a-badge--${color}`]: color,
        [`as-a-badge--${size}`]: size,
      },
      className
    )}
    data-qa={dataQA}>
    {count > max ? `${max}${COUNTER_OVERFLOW}` : `${count}`}
  </div>
);
