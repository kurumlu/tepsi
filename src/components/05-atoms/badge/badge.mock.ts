import { BadgeProps, BadgeColor, BadgeSize } from './Badge';

export const mockBadgeDefault: BadgeProps = {
  count: 5,
  max: 88,
  color: BadgeColor.SECONDARY,
  size: BadgeSize.LARGE,
};
