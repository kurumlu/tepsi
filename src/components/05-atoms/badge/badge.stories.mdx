import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { Badge, BadgeColor, BadgeSize } from './Badge';
import { Required } from '../../storybook-elements/constants/Required';

<Meta
  title="Design system/Patterns/Atoms/Badge"
  component={Badge}
/>

# Badge

> Badges show the user a count of something.
> They are typically placed next or on top of another element, and attract attention.

<Canvas>
  <Story name="Default">
    <Badge count={1} />
  </Story>
</Canvas>

---

## Available props

| Name        | Type         | Required    | Default value |
|-------------|--------------|-------------|---------------|
| `count`     | *number*     | <Required/> |               |
| `max`       | *number*     |             | *99*          |
| `color`     | *BadgeColor* |             |               |
| `size`      | *BadgeSize*  |             |               |
| `className` | *string*     |             |               |
| `dataQA`    | *string*     |             |               |

---

### Count

Use the `count` prop to set the value of the badge.

```javascript
<Badge count={...} />
```

<Canvas>
  <Story name="Count">
    <Badge count={24} />
  </Story>
</Canvas>

---

### Max

Use the `max` prop to set a maximum limit of the `count`.

```javascript
<Badge count={...} max={...} />
```

<Canvas>
  <Story name="Max">
    <Badge count={300} max={9} />
  </Story>
</Canvas>

---

### Color

Use the `color` prop to set a color.

| BadgeColor  |
|-------------|
| `SECONDARY` |
| `SUBTLE`    |

```javascript
<Badge color={...} />
```

#### Secondary

<Canvas>
  <Story name="Color - secondary">
    <Badge count={1} color={BadgeColor.SECONDARY} />
  </Story>
</Canvas>

#### Subtle

<Canvas>
  <Story name="Color - subtle">
    <Badge count={1} color={BadgeColor.SUBTLE} />
  </Story>
</Canvas>

---

### Size

Use the `size` prop to set a size.

| BadgeSize |
|-----------|
| `LARGE`   |

```javascript
<Badge size={...} />
```

#### Large

```javascript
<Badge size={BadgeSize.LARGE} />
```

<Canvas>
  <Story name="Size - large">
    <Badge count={1} size={BadgeSize.LARGE} />
  </Story>
</Canvas>

---

### Other

Other props that can be added:
- `className`
- `dataQA`
