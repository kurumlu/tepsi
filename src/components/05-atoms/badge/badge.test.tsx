import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Badge, BadgeProps, BadgeColor } from './Badge';
import { mockBadgeDefault } from './badge.mock';
import { COUNTER_OVERFLOW } from '../../00-helpers/constants/Characters';

describe('<Badge />', () => {
  const component: ShallowWrapper<BadgeProps> = shallow(
    <Badge {...mockBadgeDefault} />
  );

  it('renders <Badge /> component', () => {
    expect(component).toMatchSnapshot();
  });

  it('has default max value is 99', () => {
    const BadgeWithDefaultMaxValue: ShallowWrapper<BadgeProps> = shallow(
      <Badge count={100} />
    );
    expect(BadgeWithDefaultMaxValue).toMatchSnapshot();
    expect(BadgeWithDefaultMaxValue.text()).toBe(`99${COUNTER_OVERFLOW}`);
  });

  it(`shows max ${mockBadgeDefault.max}${COUNTER_OVERFLOW} in case count is higher then max value`, () => {
    const BadgeWithMaxCount: ShallowWrapper<BadgeProps> = shallow(
      <Badge {...mockBadgeDefault} count={100} />
    );
    expect(BadgeWithMaxCount).toMatchSnapshot();
    expect(BadgeWithMaxCount.text()).toBe(
      `${mockBadgeDefault.max}${COUNTER_OVERFLOW}`
    );
  });

  it(`adds classname 'sd-a-badge--${BadgeColor.SECONDARY}' if "color" props passed`, () => {
    expect(component.hasClass(`sd-a-badge--${BadgeColor.SECONDARY}`)).toEqual(
      true
    );
  });
});
