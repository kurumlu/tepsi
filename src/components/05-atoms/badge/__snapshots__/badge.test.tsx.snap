// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> has default max value is 99 1`] = `
<div
  className="sd-a-badge"
>
  99+
</div>
`;

exports[`<Badge /> renders <Badge /> component 1`] = `
<div
  className="sd-a-badge sd-a-badge--secondary sd-a-badge--l"
>
  5
</div>
`;

exports[`<Badge /> shows max 88+ in case count is higher then max value 1`] = `
<div
  className="sd-a-badge sd-a-badge--secondary sd-a-badge--l"
>
  88+
</div>
`;
